File: androidApp/src/main/kotlin/website/lolegrand/personapp/master/adapter/MasterAdapter.kt
package website.lolegrand.personapp.master.adapter

import website.lolegrand.personapp.local.model.GenderEntity
import website.lolegrand.personapp.local.model.PersonEntity
import website.lolegrand.personapp.master.model.MasterGenderUI
import website.lolegrand.personapp.master.model.MasterPersonUI

fun PersonEntity.toMasterPersonUI() = MasterPersonUI(
    id = id,
    name = name,
    email = email,
    profilePicUrl = pictureUrl,
    gender = gender.toMasterGenderUI(),
)

fun GenderEntity.toMasterGenderUI() = when (this) {
    GenderEntity.MALE -> MasterGenderUI.MALE
    GenderEntity.FEMALE -> MasterGenderUI.FEMALE
}

File: iosApp/iosApp/Master/Adapter/MasterAdapter.swift
//
//  MasterAdapter.swift
//  iosApp
//
//  Created by Louis Legrand on 28/01/2025.
//  Copyright © 2025 orgName. All rights reserved.
//

import Foundation
import Shared

extension GenderEntity {
    
    func toMasterGenderUI() -> MasterGenderUI {
        switch self {
        case .male: return MasterGenderUI.Male
        case .female: return MasterGenderUI.Female
        }
    }
    
}

extension [PersonEntity] {
    
    func toMasterPersonUIList() -> [MasterPersonUI] {
        return self.map { personEntity in
            personEntity.toMasterPersonUI()
        }
    }
    
}

extension PersonEntity {
    
    func toMasterPersonUI() -> MasterPersonUI {
        return MasterPersonUI(id: id, name: name, ppUrl: pictureUrl, email: email, gender: gender.toMasterGenderUI())
    }
    
}
